client/acl.go

```diff
@@ -17,9 +17,10 @@ const (
 	// construction cost, so we keep the hot objects cached to reduce the ACL token resolution time.
 	aclCacheSize = 64
 
-	// tokenCacheSize is the number of ACL tokens to keep cached. Tokens have a fetching cost,
-	// so we keep the hot tokens cached to reduce the lookups.
-	tokenCacheSize = 64
+	// tokenCacheSize is the number of bearer tokens, ACL and workload identity,
+	// to keep cached. Tokens have a fetching cost, so we keep the hot tokens
+	// cached to reduce the lookups.
+	tokenCacheSize = 128
```

Comment:
Doubled the cache size since we're now stuffing workload identities in here. No science was used to pick this number (or I suspect the previous number). The [TTL is only 30s by default](https://developer.hashicorp.com/nomad/docs/configuration/acl#token_ttl), so I suspect this is more than enough to ensure all lookups are cached and only evicted by TTL.